src/zenml/integrations/vault/secrets_manager/vault_secrets_manager.py

```diff
@@ -149,11 +79,44 @@ def _ensure_client_is_authenticated(self) -> None:
 
         if not self.CLIENT.is_authenticated():
             raise RuntimeError(
-                "There was an error authenticating with Vault. Please check your configuration."
+                "There was an error authenticating with Vault. Please check "
+                "your configuration."
             )
         else:
             pass
 
+    def _get_scoped_secret_name(self, name: str) -> str:
+        """Convert a ZenML secret name into a Vault scoped secret name.
+
+        Args:
+            name: the name of the secret
+
+        Returns:
+            The Vault scoped secret name
+        """
+        return "/".join(self._get_scoped_secret_path(name))
+
+    @staticmethod
+    def _sanitize_secret_name(secret_name: str) -> str:
+        """Sanitize the secret name to be used in Vault.
+
+        Args:
+            secret_name: The secret name to sanitize.
+
+        Returns:
+            The sanitized secret name.
+        """
+        sanitized_secret_name = re.sub(r"[^0-9a-zA-Z_\.]+", "_", secret_name)
```

Comment:
In the other secrets manager we simply prevent unallowed special characters, should we do the same thing here?

Reply:
Fully agree, should have been replaced already with https://github.com/zenml-io/zenml/pull/803/commits/82f444a3a2e58c1b92b78f2766d91de26138a34d - not sure why it's not showing up in the files diff yet

Reply:
@AlexejPenner will take care of this :)